### PR TITLE
Catch BWM crashing when jobID doesn't exist

### PR DESCRIPTION
**Pull request by mattabullock**
We already attempt to catch if the job was retried elsewhere, so that would mean the jobID doesn't exist. The error name must've changed or something...

Part of https://github.com/Expensify/Expensify/issues/95570

### CODE REVIEW

File: bin/BedrockWorkerManager.php [hunk 1]
```diff
@@ -339,7 +339,7 @@
                                 ]);
                                 try {
                                     $jobs->retryJob((int) $job['jobID'], $e->getDelay(), $worker->getData(), $e->getName(), $e->getNextRun());
-                                } catch (IllegalAction $e) {
+                                } catch (IllegalAction | DoesNotExist $e) {
```
Comment (iwiznia) [inline bin/BedrockWorkerManager.php]: If this makes sense here, it should also make sense in the other places like line 358 and 326, right?

Comment (mattabullock) [reply bin/BedrockWorkerManager.php]: We already catch it at 326, but I can add it to 358 too.

Comment (iwiznia) [reply bin/BedrockWorkerManager.php]: Oh right, did not see the other catch. I think we should add it to 358.

Comment (mattabullock) [reply bin/BedrockWorkerManager.php]: I'm trying to but github won't let me assign you the pr or edit the title.

**Pull request merged**